turbinia/workers/analysis/auth.py

```diff
@@ -0,0 +1,619 @@
+    # username statistical data collection.
+    checked_user_accounts = []
+
+    for record in brute_force_records:
+      domain = record.get('domain')
+      username = record.get('username')
+      user_account = f'{domain}_{username}'
+
+      log.debug(
+          f'[{self.NAME}] Checking for domain:{domain}, username: {username}')
+      if user_account in checked_user_accounts:
+        log.debug(
+            f'[{self.NAME}] Skipping user account {user_account}. User account'
+            f' already checked')
+        continue
+
+      user_summary = self.get_user_summary(domain=domain, username=username)
+      if not user_summary:
+        log.info(
+            f'[{self.NAME}] No user summary for domain: {domain},'
+            f' username: {username}')
+        continue
+      username_summaries.append(user_summary.report())
+
+    # Analysis report on the source_ip.
+    ip_analysis_report = {
+        'source_ip': source_ip,
+        'brute_force_logins': brute_force_records,
+        'ip_summaries': ip_summaries,
+        'user_summaries': username_summaries,
+    }
+
+    return ip_analysis_report
+
+  def generate_analyzer_output(self, reports: List, success: bool) -> dict:
+    """Generate analyzer output"""
+
+    analyzer_output = {
+        'platform': 'turbinia',
+        'analyzer_identifier': self.NAME,
+        'analyzer_name': self.DISPLAY_NAME,
+        'result_status': 'failure',
+        'dfiq_question_id': '',
+        'dfiq_question_conclusion': '',
+        'result_priority': 'LOW',
+        'result_summary': '',
+        'result_markdown': '',
+        'references': [],
+        'attributes': reports,
+    }
+
+    if not success:
+      return analyzer_output
+    analyzer_output['result_status'] = 'success'
+
+    # Generating result_summary and result_priority
+    # result_priority is set to MEDIUM for any successful brute force detection
+    reports_count = len(reports)
+    if reports_count > 0:
+      analyzer_output[
+          'result_summary'] = f'Brute force from {reports_count} IP addresses'
+      analyzer_output['result_priority'] = 'MEDIUM'
+    else:
+      analyzer_output['result_summary'] = 'No brute force detected'
+
+    # Generate result_markdown
+    markdown = []
+    markdown.append('## Brute Force Analysis\n')
+
+    for report in reports:
+      markdown.append(f'### Brute Force from {report["source_ip"]}\n')
+
+      for brute_force_login in report['brute_force_logins']:
+        markdown.append(
+            f'- Successful brute force from {brute_force_login["source_ip"]} as'
+            f' {brute_force_login["username"]} at'
+            f' {self.human_timestamp(brute_force_login["login_timestamp"])}'
+            f' (duration={brute_force_login["session_duration"]})')
+
+        if brute_force_login['session_duration'] > 600:
+          markdown.append(
+              '**NOTE**: Long login duration (>10 minutes).'
```

Comment:
I wonder if we want to find a way to increase the priority of the finding in this case.

Reply:
Fixed. Priority increased to CRITICAL when long interactive session is observed.